visualization/app/codeCharta/util/algorithm/streetLayout/strip/verticalStrip.ts

```diff
@@ -0,0 +1,58 @@
+import Strip from "./strip"
+import Rectangle from "../rectangle"
+import { Vector2 } from "three"
+import { CodeMapNode } from "../../../../codeCharta.model"
+
+export enum VerticalOrder {
+	topToBottom,
+	bottomToTop
+}
+
+export default class VerticalStrip extends Strip {
+	constructor(nodes: CodeMapNode[]) {
+		super(nodes)
+	}
+
+	layout(
+		rect: Rectangle,
+		rootSize: number,
+		metricName: string,
+		currentTreemapDepth: number,
+		margin: number,
+		order: VerticalOrder = VerticalOrder.topToBottom
+	): CodeMapNode[] {
+		let offsetY = rect.topLeft.y
+
+		const nodes = order === VerticalOrder.topToBottom ? this.nodes : [...this.nodes].reverse()
+		const rootArea = rect.area()
+		const height = rect.height
+		const width = this.totalScaledSize(nodes, metricName, rootSize, rootArea) / height
+		const stripNodes: CodeMapNode[] = []
+
+		for (const node of nodes) {
+			const nodeSize = this.scaledSize(node, rootSize, rootArea, metricName)
+			const nodeHeight = width > 0 ? nodeSize / width : 0
+			const newRect = new Rectangle(new Vector2(rect.topLeft.x, offsetY), width, nodeHeight)
+			stripNodes.push({
+				...node,
+				value: node.type === "File" ? rootSize : 0,
+				rect: this.applyNodeMargin(newRect, margin),
+				zOffset: currentTreemapDepth
+			})
+			offsetY += nodeHeight
+		}
+		return stripNodes
+	}
+
+	worstAspectRatio(nodes: CodeMapNode[], rect: Rectangle, rootSize: number, metricName: string): number {
+		const height = rect.height
+		const rootArea = rect.area()
+		const totalSize = this.totalScaledSize(nodes, metricName, rootSize, rootArea)
+		const stripMin = this.min(nodes, metricName, rootSize, rootArea)
+		const stripMax = this.max(nodes, metricName, rootSize, rootArea)
+		const heightSquared = Math.pow(height, 2)
+		const totalSizeSquared = Math.pow(totalSize, 2)
+
+		return Math.max((heightSquared * stripMax) / totalSizeSquared, totalSizeSquared / (heightSquared * stripMin))
+	}
+}
```

Comment:
These function seem very similar to some other functions. Maybe they could be abstracted a bit?

Reply:
I think there are a lot of refactoring that should be done, just wanted to push and we might refactor it later.